dhis-2/dhis-api/src/main/java/org/hisp/dhis/util/SharingUtils.java

```diff
@@ -75,6 +120,12 @@ public static Sharing generateSharingFromIdentifiableObject( IdentifiableObject
         return sharing;
     }
 
+    public static void resetAccessCollections( BaseIdentifiableObject identifiableObject )
+    {
+        identifiableObject.setUserAccesses( Sets.newHashSet() );
+        identifiableObject.setUserGroupAccesses( Sets.newHashSet() );
```

Comment:
I'd think we should be fine with `emptySet()` here as well or do we have code that manipulates these collection instead of setting a new one?

Reply:
this is used by the importing services, those collections will go through the processes of preheatService, mergeService and other related objectBundleHooks. So i think it safe to create new empty set rather than return an immutable empty set.